### ISSUE TITLE
make pyenv install correctly with pip

### ISSUE THREAD

**Issue by ssbarnea** [pyenv/pyenv-installer#43]
The fact that there is a `pyenv` package on `pypi` that is outdated and that installs incorrectly. This is causing confusions for everyone.

`pip` became the official way to install Python software, is easy to use and reliable. I think is used by 95% of python packages so why having a broken version on pypi? If Is broken why is still there?

Is there a bug in current versions of pip that prevents proper distribution of pyenv via it?

**Comment by obestwalter** [comment]
I agree that the situation is somehow awkward/unsatisfactorily. I must admit that I dropped the ball there, as I initiated this and am not following it through properly. 

The problem is that pyenv is not a Python package itself, but a collection of shell scripts. The reason pyenv is on PyPi at all was that there used to be an unmaintained package on PyPi with the same name that had nothing to do with the pyenv project. I asked the maintainer of the PyPi project if he plans to still work on that or if he would like to free the package so that pyenv could be also found through Pypi and made a proof of concept for Python2 to install it through Pypi. The PoC was merged and the package was added to PyPi and that was basically it. My main aim at that time was that the name pyenv would be clearly connected to this project, instead of pointing to a defunct project from PyPi. 

In my defense though it is stated clearly in the docs  that the PyPi way of installing is a hack and that it is not recommended to install it through PyPi, but I am unhappy with the way it is myself, so it's good that you bring it up.

I would also like it if it would work well with pip. I have not yet thought about how to do that elegantly. So ideas how to that properly would be very welcome.

Maybe it would be a good idea to have a different package on PyPi that clearly explains the special status of pyenv and is a wrapper for the "real" pyenv installer?

**Comment by ssbarnea** [comment]
@obestwalter Thank you for clarifying this. As I have some experience dealing with PIP and PYPI, I could try to give a hand and maybe even if this issue once for all.

**Comment by obestwalter** [comment]
@ssbarnea That would be great! I did this quite a while ago and I did not have that much insight into packaging, etc. as I have today, so I expect what I did was pretty gruesome anyway.

Disclaimer: I have a lot on my plate at the moment, but I will try to carve out some time to look into it if you pick this up now and want to help or even create a PR or even only discuss what a good enough solution would be. 

Would be great to remove that HACK warning and announce that pyenv is working properly with pip :)
